### PR TITLE
created ImageMaskFromImage and LoadTextureMaskFromImage

### PR DESCRIPTION
**Pull request by brccabral**
These functions adds the possibility to retrieve a mask from the alpha channel of an image.

### CODE REVIEW

File: src/rtextures.c [hunk 1]
```diff
@@ -3979,22 +4017,29 @@ Texture2D LoadTexture(const char *fileName)
 
 // Load a texture from image data
 // NOTE: image is not unloaded, it must be done manually
-Texture2D LoadTextureFromImage(Image image)
-{
-    Texture2D texture = { 0 };
+Texture2D LoadTextureFromImage(Image image) {
+  Texture2D texture = {0};
```
Comment (asdqwe) [inline src/rtextures.c:4021]: Hello,
Please revert the changes on `LoadTextureFromImage()` as it breaks current [code conventions](https://github.com/raysan5/raylib/wiki/raylib-coding-conventions).
Best regards.

File: src/rtextures.c [hunk 2]
```diff
@@ -3979,22 +4017,29 @@ Texture2D LoadTexture(const char *fileName)
...
-    if ((image.width != 0) && (image.height != 0))
-    {
-        texture.id = rlLoadTexture(image.data, image.width, image.height, image.format, image.mipmaps);
-    }
-    else TRACELOG(LOG_WARNING, "IMAGE: Data is not valid to load texture");
+  if ((image.width != 0) && (image.height != 0)) {
+    texture.id = rlLoadTexture(image.data, image.width, image.height,
+                               image.format, image.mipmaps);
+  } else
+    TRACELOG(LOG_WARNING, "IMAGE: Data is not valid to load texture");
```
Comment (asdqwe) [inline src/rtextures.c:4027]: Hello,
Please revert the changes on `LoadTextureFromImage()` as it breaks current [code conventions](https://github.com/raysan5/raylib/wiki/raylib-coding-conventions).
Best regards.

File: src/rtextures.c [hunk 3]
```diff
@@ -3979,22 +4017,29 @@ Texture2D LoadTexture(const char *fileName)
...
 
-    texture.width = image.width;
-    texture.height = image.height;
-    texture.mipmaps = image.mipmaps;
-    texture.format = image.format;
+  texture.width = image.width;
+  texture.height = image.height;
+  texture.mipmaps = image.mipmaps;
+  texture.format = image.format;
```
Comment (asdqwe) [inline src/rtextures.c:4032]: Hello,
Please revert the changes on `LoadTextureFromImage()` as it breaks current [code conventions](https://github.com/raysan5/raylib/wiki/raylib-coding-conventions).
Best regards.

File: src/rtextures.c [hunk 4]
```diff
@@ -3979,22 +4017,29 @@ Texture2D LoadTexture(const char *fileName)
...
 
-    return texture;
+  return texture;
```
Comment (asdqwe) [inline src/rtextures.c:4034]: Hello,
Please revert the changes on `LoadTextureFromImage()` as it breaks current [code conventions](https://github.com/raysan5/raylib/wiki/raylib-coding-conventions).
Best regards.

File: src/raylib.h [hunk 5]
```diff
@@ -1334,6 +1334,7 @@ RLAPI Image ImageCopy(Image image);
 RLAPI Image ImageFromImage(Image image, Rectangle rec);                                                  // Create an image from another image piece
 RLAPI Image ImageText(const char *text, int fontSize, Color color);                                      // Create an image from text (default font)
 RLAPI Image ImageTextEx(Font font, const char *text, float fontSize, float spacing, Color tint);         // Create an image from text (custom sprite font)
+RLAPI Image ImageMaskFromImage(Image image, Color color, float threshold);                               // Create an image mask from an image (alpha channel)
```
Comment (raysan5) [inline src/raylib.h:1337]: I think this function could be just `ImageFromChannel()` and return selected channel as a GRAYSCALE/R32 image.

File: src/raylib.h [hunk 6]
```diff
@@ -1394,6 +1395,7 @@ RLAPI void ImageDrawTextEx(Image *dst, Font font, const char *text, Vector2 posi
 // NOTE: These functions require GPU access
 RLAPI Texture2D LoadTexture(const char *fileName);                                                       // Load texture from file into GPU memory (VRAM)
 RLAPI Texture2D LoadTextureFromImage(Image image);                                                       // Load texture from image data
+RLAPI Texture2D LoadTextureMaskFromImage(Image image, Color color, float threshold);                            // Load texture mask from image data (alpha channel)
```
Comment (raysan5) [inline src/raylib.h:1398]: I don't think this function is actually needed. Just with `ImageFromChannel()` + `LoadTextureFromImage()` the same result could be acomplished. I prefer to minimize raylib functions.

File: src/raylib.h (hunk 5 above)
Comment (brccabral) [reply src/raylib.h:1337]: Can't find ImageFromChannel in the master branch...

Comment (raysan5) [reply src/raylib.h:1337]: @brccabral It does not exist, it's a proposed more general alternative to adding `ImageMaskFromImage()`

Comment (brccabral) [reply src/raylib.h:1337]: Got it. Close this PR whenever you want to. Looking forward to ImageFromChannel. Saludos
